core/engine/src/builtins/function/mod.rs

```diff
@@ -850,6 +850,13 @@ impl BuiltInFunctionObject {
             );
         } else if object_borrow.is::<Proxy>() || object_borrow.is::<BoundFunction>() {
             return Ok(js_string!("function () { [native code] }").into());
+        } else if object_borrow.is::<LazyBuiltIn>() {
+            let name = object_borrow
+                .downcast_ref::<LazyBuiltIn>()
+                .map_or_else(|| js_string!(), |built_in| built_in.name.clone());
+            return Ok(
+                js_string!(js_str!("function "), &name, js_str!("() { [native code] }")).into(),
+            );
```

Comment:
This is not needed if you move the check for `LazyBuiltIn` to the same branch as the check for `NativeFunctionObject`. The reason is that we call `JsObject::get` inside it, which will already initialize the `name` of the builtin, printing the correct result.